StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx

```diff
@@ -17,13 +19,13 @@ ClassImp(StrangeMuDstPlayer)
 
 char empty = '\0';
 char slash = '/';
-void ParseFileName(Char_t *output, Char_t **file, Char_t **dir) {
-  if ((*file = strrchr(output,slash))) {
+void ParseFileName(const Char_t *output, Char_t **file, Char_t **dir) {
```

Comment:
```suggestion
void ParseFileName(const Char_t *output, const Char_t **file, Char_t **dir) {
```

Reply:
and s/`Char_t *file, *dir, *outfile[3], line[80];`/`const Char_t *file, *dir; Char_t *outfile[3], line[80];`/

Reply:
Looks like the output argument. You think it will compile?

Reply:
P.S. C pointer types are to be read right to left
https://cdecl.org/?q=declare+x+as+pointer+to+pointer+to+const+char -> `const char **x`
https://cdecl.org/?q=declare+x+as+pointer+to+const+pointer+to+char -> `char * const *x`

Reply:
I tried your suggestion but it did not work for me:

```
.sl79_gcc485/OBJ/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx: In member function 'void StrangeMuDstPlayer::Make(Int_t, StFile*, const Char_t*)':
.sl79_gcc485/OBJ/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx:117:38: error: invalid conversion from 'const Char_t** {aka const char**}' to 'Char_t** {aka char**}' [-fpermissive]
     ParseFileName(output, &file, &dir);
                                      ^
.sl79_gcc485/OBJ/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx:22:6: error:   initializing argument 3 of 'void ParseFileName(const Char_t*, const Char_t**, Char_t**)' [-fpermissive]
 void ParseFileName(const Char_t *output, const Char_t **file, Char_t **dir) {
      ^
.sl79_gcc485/OBJ/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx:121:23: error: invalid conversion from 'const Char_t* {aka const char*}' to 'char*' [-fpermissive]
       strcpy(outfile[i],dir);
                       ^
```

Reply:
@plexoos Could you also post your diff?

Reply:
Okay, had to make dir constant too and massage it's creation (memory leaks preserved!).
```diff
diff --git a/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx b/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx
index 349fea08b7..a6a9c61dd7 100755
--- a/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx
+++ b/StRoot/StStrangeMuDstMaker/StStrangeMuDstPlayer.cxx
@@ -19,13 +19,14 @@ ClassImp(StrangeMuDstPlayer)
 
 char empty = '\0';
 char slash = '/';
-void ParseFileName(const Char_t *output, Char_t **file, Char_t **dir) {
-  if ((*file = std::strrchr(const_cast<char*>(output),slash))) {
+void ParseFileName(const Char_t *output, const Char_t **file, const Char_t **dir) {
+  if ((*file = std::strrchr(output,slash))) {
     (*file)++;
-    *dir = new char[strlen(output)+5];
-    strncpy(*dir,output,(strlen(output)-strlen(*file)));
+    char *_dir = new char[strlen(output)+5];
+    strncpy(_dir,output,(strlen(output)-strlen(*file)));
+    *dir = _dir;
   } else {
-    *file = const_cast<char*>(output);
+    *file = output;
     *dir = &empty;
   }
 }
@@ -64,7 +65,7 @@ void StrangeMuDstPlayer::Make(Int_t NEvents, StFile* input, const Char_t* output
   StMcEventMaker *mcEventReader;
   StAssociationMaker *associator;
 //   StRandyTopMapMaker *topoMapFixer;  // RWitt: Commented 7/24/15 to remove dependency on StHbtMaker
-  Char_t *file, *dir, *outfile[3], line[80];
+  const Char_t *file, *dir; Char_t *outfile[3], line[80];
   TString prefix[3];
   Int_t mNDstMakers = 0;
 
@@ -175,7 +176,7 @@ void StrangeMuDstPlayer::Filter(Int_t NEvents, StFile* input, const Char_t* outp
   StStrangeMuDstMaker *v0MuDstMaker=0;
   StStrangeMuDstMaker *xiMuDstMaker=0;
   StStrangeMuDstMaker *kinkMuDstMaker=0;
-  Char_t *file, *dir, *outfile[3], line[80];
+  const Char_t *file, *dir; Char_t *outfile[3], line[80];
   TString prefix[3];
   Int_t mNDstMakers = 0;
 
@@ -380,7 +381,7 @@ void StrangeMuDstPlayer::Play(Int_t NEvents, StFile* input, const Char_t* output
   StMcEventMaker *mcEventReader;
   StAssociationMaker *associator;
   //   StRandyTopMapMaker *topoMapFixer; // RWitt: Commented 7/24/15 to remove dependency on StHbtMaker
-  Char_t *file, *dir, *outfile[3], line[80];
+  const Char_t *file, *dir; Char_t *outfile[3], line[80];
   TString prefix[3];
   Int_t mNDstMakers = 0;
 
@@ -601,7 +602,7 @@ void StrangeMuDstPlayer::Copy(Int_t NEvents, StFile* input, const Char_t* output
   StStrangeMuDstMaker *v0MuDstMaker=0;
   StStrangeMuDstMaker *xiMuDstMaker=0;
   StStrangeMuDstMaker *kinkMuDstMaker=0;
-  Char_t *file, *dir, *outfile[3], line[80];
+  const Char_t *file, *dir; Char_t *outfile[3], line[80];
   TString prefix[3];
   Int_t mNDstMakers = 0;
 
```

Reply:
Very nice! Much better than the const cast
Do you want to commit this patch?